### DISCUSSION TITLE
How can I add a compile definition to a subdirectory?

### DISCUSSION THREAD

**Discussion by xtremegamer1** [build-cpp/cmkr#76] [Q&A]
I imagine it should look something like this:
```
[subdir.asmjit]
compile_definitions = [
    "ASMJIT_STATIC"
]
```

**Comment by gmh5225** [comment]
Maybe you could try it this way
```
[subdir.asmjit]
include-before = ["asmjit-static.cmake"]
```

In asmjit-static.cmake:
`add_definitions(-DASMJIT_STATIC)`

**Comment by mrexodia** [comment]
Yeah you can do this, but it’s not the correct way to do this. Asmjit has a CMake `option` to build statically and you should set that instead:

```toml
[subdir.asmjit]
cmake-before = "set(ASMJIT_STATIC ON)"
```

The `DEFINED` keyword in CMake (confusingly) has _nothing_ to do with compile definitions and instead checks if a CMake variable is defined.
